common/src/primitives/id/with_id.rs

```diff
@@ -78,12 +79,19 @@ impl<T: Idable> From<T> for WithId<T> {
     }
 }
 
+impl<T: Idable> WrapperTypeDecode for WithId<T> {
```

Comment:
Why would we need to serialize and deserialize WithId? We shouldn't be storing serialized objects with their hashes.

This is actually a recipe for bugs. A scenario for this is as follows: We serialize something, break the invariant and modify the hash, then deserialize it. 

Why not deserialize objects then put them in WithId?

Reply:
The `WrapperTypeEncode` does not encode the hash. For example in the `serialization()` test in `with_id.rs` encodes this struct

```
TestStruct {
    num: 1337,
    blurb: "Hello!".into(),
}
```

as `[39, 05, 00, 00, 00, 00, 00, 00, 18, 48, 65, 6C, 6C, 6F, 21]`.

Here is the relevant source of that trait:

```
/// A marker trait for types that wrap other encodable type.
///
/// Such types should not carry any additional information
/// that would require to be encoded, because the encoding
/// is assumed to be the same as the wrapped type.
///
/// The wrapped type that is referred to is the [`Deref::Target`].
pub trait WrapperTypeEncode: Deref {}
```

Reply:
I see. That's fair. Then please add it to the `with_id.rs` file instead of here. Then we can scrutinize this as a generic solution instead of just for the wallet.

Reply:
Added some comments. As it is now, anywhere `WithId<T>` is used it can be serialised as if `T` was serialised.

Reply:
Can you please move this to `with_id.rs` file and import it from there?

Reply:
I am not sure what do you mean by moving this to `with_id.rs`. The impl is already in `with_id.rs` but there is nothing to import.